standard/page.lua

```diff
@@ -0,0 +1,61 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Page
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local String = require('Module:StringUtils')
+
+local page = {}
+
+function page.exists(page)
+	local existingPage = mw.title.new(page)
+
+	-- In some cases we might have gotten an external link,
+	-- which will mean `existingPage` will equal nil
+	if existingPage == nil then
+		return false
+	end
+
+	return existingPage.exists
+end
+
+function page.makeInternalLink(link, display)
+	if String.isEmpty(link) then
+		return nil
+	elseif String.isEmpty(display) then
+		display = link
+	end
+
+	return '[[' .. link .. '|' .. display .. ']]'
+end
+
+function page.makeInternalLinkIfExists(link, display)
+	if String.isEmpty(link) then
+		return nil
+	elseif String.isEmpty(display) then
+		display = link
+	end
+	if not page.exists(link) then
+		return nil
+	end
+
+	return '[[' .. link .. '|' .. display .. ']]'
+end
```

Comment:
could combine those via
```suggestion
function page.makeInternalLink(link, display, options)
	if String.isEmpty(link) then
		return nil
	elseif (options or {}).ifExists == true and (not page.exists(link)) then
		return nil
	end
	if String.isEmpty(display) then
		display = link
	end

	return '[[' .. link .. '|' .. display .. ']]'
end
```
but then we could not use it from wikicode directly with the option
question if we need to be able to do that

Reply:
If you put options first, you could use it via `Class.export()`, even from wikicode. Might be best option?

Reply:
would need to add a catch against the case of empty options

Reply:
Sure, not super ideal, but worth the tradeoff I think?